lambeq/training/pytorch_model.py

```diff
@@ -37,11 +43,31 @@ class PytorchModel(Model, torch.nn.Module):
 
     weights: torch.nn.ParameterList  # type: ignore[assignment]
     symbols: list[Symbol]
+    tn_path_optimizer: TnPathOptimizer
 
-    def __init__(self) -> None:
+    def __init__(
+            self,
+            tn_path_optimizer: TnPathOptimizer | None = None
+    ) -> None:
         """Initialise a PytorchModel."""
         Model.__init__(self)
         torch.nn.Module.__init__(self)
+        self.tn_path_optimizer = (
```

Comment:
Perhaps it would be better if the path optimiser was not so much coupled with the code of of the model here, e.g. (a) could be activated/deactivated by an argument on the constructor (b) creation of instance and use will be conditional to the flag. It could stay activated by default, but it wouldn't do anything at all if not.

Reply:
I see. But let's at least rename the checkpoint methods in the path optimisers to:

- `store_to_checkpoint()`
- `retrieve_from_checkpoint()`

or something similar to avoid confusion

Reply:
Just had a second look and looks you're right, let's keep it like this.